services-hll/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/ConverterAware.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.model;
+
+import java.util.Collection;
+import java.util.List;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.NotThreadSafe;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
+import software.amazon.awssdk.enhanced.dynamodb.converter.DefaultConverterChain;
+import software.amazon.awssdk.enhanced.dynamodb.converter.ItemAttributeValueConverter;
+
+/**
+ * An interface applied to all objects that wish to expose their underlying {@link ItemAttributeValueConverter}s.
+ *
+ * See {@link ItemAttributeValueConverter} for a detailed explanation of how the enhanced client converts between Java types
+ * and DynamoDB types.
+ */
+@SdkPublicApi
+@ThreadSafe
+@Immutable
+public interface ConverterAware {
+    /**
+     * Retrieve all converters that were directly configured on this object.
+     */
+    List<ItemAttributeValueConverter> converters();
```

Comment:
Should we use `Collection` instead?

Reply:
`Collection` isn't necessarily ordered. I wanted to use `List` to indicate that order mattered.